### ISSUE TITLE
[GNOME 46] History Graph in drop down not showing

### ISSUE THREAD

**Issue by 3dsf** [AstraExt/astra-monitor#40]
## Description
The drop down does not show a history graph even though there is activity 

## Steps to Reproduce

1. Unknown 

## Screenshots

![Screenshot from 2024-01-25 19-53-41](https://github.com/AstraExt/astra-monitor/assets/44537624/a4bed9ab-b838-45e4-955f-9332077f1dec)

## Environment

- **Astra Monitor Version**: [pre-v8](https://github.com/AstraExt/astra-monitor/issues/38#issuecomment-1908998274)
- **GNOME Version**: 46
- **Operating System and Version**: Fedora Rawhide (6.8)
- **Other Relevant System Information**:

## Additional Context

Checked, persists with a reboot
Also, you ko-fi page is not fully enabled

**Comment by ljuzig** [comment]
Thanks for the submission. I think I'll wait a bit to see this kind of bug. GNOME 46 is in alpha release, and according to the EGO team, a lot of features are prone to bugs and changes in the coming weeks. To better figure out what's on my end and what's on GNOME's, I'll wait for the feature freeze with the beta release, which I believe is on February 10th. For now, I'll keep this issue open, but I'll take another look if the problem persists after the February 10th beta.

As you can see from this screen a lot of things do look very different in Gnome 46 right now, not just the graph:
![image](https://github.com/AstraExt/astra-monitor/assets/11982322/13e96701-a0c4-4397-a530-be576fa88a4b)

**Comment by ljuzig** [comment]
Also try to update to the version on [EGO](https://extensions.gnome.org/extension/6682/astra-monitor/) which have a lot of fixes over your version.
Oh and many thanks for the ko-fi page. I fixed it!

**Comment by 3dsf** [comment]
I ran `journalctl -b -e` for something unrelated and saw a bunch of entries like this

```
Jan 26 11:30:24 fedora gnome-shell[2723]: JS ERROR: TypeError: (intermediate value).cairo_set_source_color is not a function
                                          repaint@file:///home/3dsf/.local/share/gnome-shell/extensions/monitor@astraext.github.io/src/processor/processorGraph.js:78:17
                                          GraphBase/<@file:///home/3dsf/.local/share/gnome-shell/extensions/monitor@astraext.github.io/src/graph.js:62:57
                                          setUsageHistory@file:///home/3dsf/.local/share/gnome-shell/extensions/monitor@astraext.github.io/src/graph.js:162:27
                                          buildGraph/<@file:///home/3dsf/.local/share/gnome-shell/extensions/monitor@astraext.github.io/src/processor/processorHeader.js:242:24
                                          notify@file:///home/3dsf/.local/share/gnome-shell/extensions/monitor@astraext.github.io/src/monitor.js:175:26
                                          @resource:///org/gnome/shell/ui/init.js:21:20

```

**Comment by ljuzig** [comment]
@3dsf This is related and the very likely the root cause of the missing graphs. It's been [deprecated](https://docs.gtk.org/gdk3/func.cairo_set_source_color.html) for a while so probably they just removed it. I will fix that soon and give you a build to test if it fixes the graphs.

**Comment by ljuzig** [comment]
@3dsf try this version, the graphs rendering should be fixed now:
[monitor@astraext.github.io.shell-extension.zip](https://github.com/AstraExt/astra-monitor/files/14072339/monitor%40astraext.github.io.shell-extension.zip)

As for the layout bugs, I need more time to fix that. I have to completely replace an important component used almost everywhere.

**Comment by ljuzig** [comment]
I actually fixed the layout problems without replacing Clutter. We keep it for now since there's no proper replacement of Clutter.GridLayout on St.
I tested graphs on my Gnome 46 build and they are still not working but the error is gone. That's very strange, I need to ask for feedback of the gnome community for this.

Updated package:
[monitor@astraext.github.io.shell-extension.zip](https://github.com/AstraExt/astra-monitor/files/14072591/monitor%40astraext.github.io.shell-extension.zip)
